### ISSUE TITLE
Rule's Options Schema

### ISSUE THREAD

**Issue by azu** [textlint/textlint#700]
Currently, textlint does not check the actual option.

So, the user pass an invalid option and does not show any errors.

For example, [textlint-rule-en-capitalization](https://github.com/textlint-rule/textlint-rule-en-capitalization) support folllowing options.

```jsonc
{
    "rules": {
        "en-capitalization": {
             // allow lower-case words in Header
             "allowHeading": true,
             // allow lower-case words in Image alt
             "allowFigures": true,
             // allow lower-case words in ListItem
             "allowLists": true,
             // allow lower-case words in anywhere
             "allowWords": []
         }
    }
}
```

However, the user passed an invalid option like `"allowHead: true` and textlint does not show any error.

```jsonc
{
    "rules": {
        "en-capitalization": {
             "allowHead": true
         }
    }
}
```

textlint does not know that the rule has what options.

## Rule Options Schema

To resolve this, we need to introduce rule's options schema.
ESLint has similar one - [Options Schemas](https://eslint.org/docs/developer-guide/working-with-rules#options-schemas-1).
Each rule defines JSON Schema and ESLint use it for validation.

As a various rule author, I do not want to write JSON Schema by handwriting 😞 
We can write a rule in TypeScript and We can use Type for JSON Schema.

I've created [babel-plugin-textlint-scripts](https://github.com/textlint/babel-plugin-textlint-scripts).
This babel plugin converts TypeScript type to JSON Schema.

**In**

```ts
export type Options = {
    str_key: string;
};
const report = () => {
    return {};
};
export default report;
```

**Out**

```js
const report = () => {
  return {};
};

export default report;
export const meta = {
  name: "@textlint/babel-plugin-textlint-scripts",
  description: "A babel plugin for textlint-scripts.",
  homepage: "https://github.com/textlint/babel-plugin-textlint-scripts",
  schema: {
    "type": "object",
    "properties": {
      "str_key": {
        "type": "string"
      }
    },
    "additionalProperties": false,
    "required": ["str_key"],
    "$schema": "http://json-schema.org/draft-07/schema#"
  }
};
```

This plugin will help us to integrate Options schema.

Of course, we can write JSON  by handwriting. 

## TODO

- [ ] Define spec for JSON schema and `meta` object
- [ ] Implement validation logics into `textlint`
  - Should we implment this into `@textlint/core`?

**Comment by azu** [comment]
Recently, I've created [@textlint/config\-loader](https://github.com/textlint/editor/tree/master/packages/@textlint/config-loader) package.
We create a new `@textlint/config-validator` package that validates the rule option with the config.
